src/development/variables.md

```diff
@@ -4,7 +4,9 @@ Platform.sh allows a high degree of control over both the build process and the
 
 ## Types of variables
 
-There are three different types of variable: User-provided project variables, user-provided environment variables, and Platform.sh-provided environment variables.  All of those may be simple strings or base64-encoded JSON-serialized values.
+There are four different types of variable: User-provided project variables, user-provided environment variables, application-provided variables, and Platform.sh-provided environment variables.  All of those may be simple strings or base64-encoded JSON-serialized values.
```

Comment:
The first User is because it's the first character after a colon, thus beginning a new sentence.